File: python/day02/src/main.py
rps = {
    "A X": (4, 3), "A Y": (8, 4), "A Z": (3, 8),
    "B X": (1, 1), "B Y": (5, 5), "B Z": (9, 9),
    "C X": (7, 2), "C Y": (2, 6), "C Z": (6, 7),
}

def main():
    input = [l.strip() for l in open("../input.txt").readlines()]
    print(f"Silver: {silver(input)}")
    print(f"Gold: {gold(input)}")

def silver(input):
    return sum(rps[x][0] for x in input)

def gold(input):
    return sum(rps[x][1] for x in input)

main()

File: java/day02/src/Main.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.List;
import java.util.Map;

class Main
{ 
    public static int silver(List<String> input) {
        int sum = 0;
        Map<String, Integer> rps = Map.of(
            "A X", 4, "A Y", 8, "A Z", 3,
            "B X", 1, "B Y", 5, "B Z", 9,
            "C X", 7, "C Y", 2, "C Z", 6
        );
        for (String line: input) {
            sum += rps.get(line);
        }
        return sum;
    }

    public static int gold(List<String> input) {
        int sum = 0;
        Map<String, Integer> rps = Map.of(
            "A X", 3, "A Y", 4, "A Z", 8,
            "B X", 1, "B Y", 5, "B Z", 9,
            "C X", 2, "C Y", 6, "C Z", 7
        );
        for (String line: input) {
            sum += rps.get(line);
        }
        return sum;
    }

    public static void main(String[] args) throws IOException {
        List<String> input = Files.readAllLines(Paths.get("../input.txt"));
        System.out.println("Silver: " + silver(input));
        System.out.println("Gold: " + gold(input));
    }
}